src/domain/dataverse/domain.ts

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable max-lines-per-function */
+import { pipe } from 'fp-ts/function'
+import * as A from 'fp-ts/Array'
+import * as O from 'fp-ts/Option'
+import * as T from 'fp-ts/Task'
+import * as TE from 'fp-ts/TaskEither'
+import { immer } from 'zustand/middleware/immer'
+import { devtools } from 'zustand/middleware'
+import type { StoreApi } from 'zustand/vanilla'
+import { createStore } from 'zustand/vanilla'
+import type { Dataverse } from './entity'
+import type { DataversePort } from './port'
+import type { IO } from 'fp-ts/lib/IO'
+import type { ForgetType } from '@/util/type'
+import type { ByTypeFilterInput, Command, DataverseElementType } from './command'
+import type { ByTypeQueryFilter, DataverseQuery } from './valueObject'
+import type { Query } from './query'
+
+export type State = {
+  data: Dataverse & DataverseQuery
+}
+
+type Domain = State & Command & Query
+export type DomainAPI = ForgetType<State, Domain>
+
+export type Options = {
+  initialState: State
+}
+
+const isAllByTypeFilter = (filter: ByTypeQueryFilter): filter is 'all' => filter === 'all'
+
+const removeByTypeFilter = (
+  input: DataverseElementType,
+  filter: DataverseElementType[]
+): ByTypeQueryFilter =>
+  pipe(
+    filter,
+    O.fromPredicate(l => l.length > 1),
+    O.matchW(
+      () => 'all',
+      A.filter(r => r !== input)
+    )
+  )
+
+const addByTypeFilter = (
+  input: DataverseElementType,
+  filter: DataverseElementType[]
+): DataverseElementType[] => A.append(input)(filter)
```

Comment:
In its curried form it would simplify the code:

```suggestion
const addByTypeFilter = A.append
```